src/FlubuCore/Context/Extensions/ContextPropertiesExtensions.cs

```diff
@@ -1 +1,295 @@
        /// <returns></returns>
        public static VSSolution GetVsSolution(this IBuildPropertiesContext context, string solutionFileName = null)
        {
            var taskContext = context as ITaskContext;
            if (taskContext == null)
            {
                return null;
            }

            return ContextBaseExtensions.GetVsSolution(taskContext, solutionFileName);
        }

        /// <summary>
        /// Gets all files matching glob pattern.
        /// See: https://github.com/kthompson/glob for supported pattern expressions and use cases.
        /// </summary>
        /// <param name="context"></param>
        /// <param name="directory"></param>
        /// <param name="globPattern"></param>
        /// <returns></returns>
        public static List<FileFullPath> GetFiles(this IBuildPropertiesContext context, string directory,  params string[] globPattern)
        {
            return ContextBaseExtensions.GetFiles(directory, globPattern);
        }

        /// <summary>
        /// Gets all files matching glob pattern.
        /// See: https://github.com/kthompson/glob for supported pattern expressions and use cases.
        /// </summary>
        /// <param name="context"></param>
        /// <param name="directory"></param>
        /// <param name="globPattern"></param>
        /// <returns></returns>
        public static List<FileFullPath> GetFiles(this IBuildPropertiesContext context, string directory, GlobOptions globOptions = GlobOptions.None, params string[] globPattern)
        {
           return ContextBaseExtensions.GetFiles(directory, globOptions, globPattern);
        }

        /// <summary>
        /// Gets all directories matching glob pattern.
        /// </summary>
        /// <param name="context"></param>
        /// <param name="directory"></param>
        /// <param name="globPattern"></param>
        /// <returns></returns>
        public static List<FullPath> GetDirectories(this IBuildPropertiesContext context, string directory, params string[] globPattern)
        {
           return ContextBaseExtensions.GetDirectories(directory, GlobOptions.None, globPattern);
        }

        /// <summary>
        /// Gets all directories matching glob pattern.
        /// </summary>
        /// <param name="context"></param>
        /// <param name="directory"></param>
        /// <param name="globPattern"></param>
        /// <returns></returns>
        public static List<FullPath> GetDirectories(this IBuildPropertiesContext context, string directory, GlobOptions globOptions = GlobOptions.None, params string[] globPattern)
        {
            return ContextBaseExtensions.GetDirectories(directory, globOptions, globPattern);
        }

        public static FullPath RootDirectory(this IBuildPropertiesContext context)
        {
            return new FullPath(context.Properties.GetProductRootDir());
        }
    }
}
\ No newline at end of file
+﻿using System;
```

Comment:
It was CR at each end of line, hence the `^M` characters. 
Other files uses CRLF as EOL, so I replaced CR with CRLF.